### PR TITLE
Fixes error handling when adding an erroneous WMS URL

### PR DESCRIPTION
**Pull request by javierarce**
Updates the WMS pane so it correctly captures the error response from the WMS proxy server ( = a 200 OK response with a 'error' variable in the body)

This PR also adds a couple of new specs to test the success and the error responses.

### CODE REVIEW

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1769/))

**Comment by javierarce** [comment]
retest this please

**Comment by javierarce** [comment]
retest this please

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1790/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1823/))

**Comment by javierarce** [comment]
@CartoDB/frontend: can you review [this PR](https://github.com/CartoDB/cartodb/pull/2334/files), please?

File: lib/assets/test/spec/cartodb/table/basemap_wms_dialog.spec.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    });
+
+    var url = "I'm not an URL, you know";
+
+    view.checkTileJson(url);
+
+    setTimeout(function() {
+      expect(error).toEqual(true);
+      done();
+    }, 500);
```
Comment (viddo) [inline lib/assets/test/spec/cartodb/table/basemap_wms_dialog.spec.js]: I really think we should not do this kind timeouts in the tests. They are not deterministic (if a test for whatever reasons takes longer than the timeout), and accumulated tests like this slow down the test suite.

In this case, simply put the expectation and done() call inside the bind method and the test will timeout if it fails ([override the default timeout time](http://jasmine.github.io/2.1/introduction.html#section-Asynchronous_Support) if you consider it too long).

Comment (javierarce) [inline lib/assets/test/spec/cartodb/table/basemap_wms_dialog.spec.js]: You're totally right. I've fixed it.

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1824/))

**Pull request merged**